### ISSUE TITLE
below, the outer `with device` block causes the current device 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#4238]
stack to be cleared. The nested ones use a `lambda` to set the desired
device and ignore any devices that may be set by the custom getter.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [693b339ab2f062ec5bbb29f976c5d1fd94fbffa5](https://github.com/imdone/tensorflow/commit/693b339ab2f062ec5bbb29f976c5d1fd94fbffa5) by Francois Chollet fchollet@google.com*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/python/keras/_impl/keras/layers/normalization.py#L334-L341
